drivers/api/no_os_spi.c

```diff
@@ -138,11 +140,15 @@ void no_os_spibus_remove(uint32_t bus_number)
 	struct no_os_spibus_desc *spi_bus_desriptor = (struct no_os_spibus_desc *)
 			spi_table[bus_number];
 
-	no_os_mutex_remove(spi_bus_desriptor->mutex);
+	spi_bus_desriptor->slave_number = spi_bus_desriptor->slave_number - 1;
```

Comment:
you can simplify by doing `spi_bus_desriptor->slave_number--;`

Reply:
also, since we're here, you should only do `slave_number--` if `slave_number > 0`, otherwise you can end up with a wrap around

Reply:
these comments are valid both for spi and i2c